lib/graphql/client.ex

```diff
@@ -29,21 +34,26 @@ defmodule BridgeEx.Graphql.Client do
   @spec call(
           url :: String.t(),
           query :: String.t(),
-          variables :: map() | String.t(),
-          http_options :: Keyword.t(),
-          http_headers :: map(),
-          retry_options :: Keyword.t(),
-          log_options :: Keyword.t()
+          variables :: map(),
+          opts :: Keyword.t()
         ) :: bridge_response()
   def call(
         url,
         query,
         variables,
-        http_options,
-        http_headers,
-        retry_options,
-        log_options
+        opts
       ) do
+    encode_variables = Keyword.get(opts, :encode_variables)
+    http_options = Keyword.get(opts, :http_options)
+    http_headers = Keyword.get(opts, :http_headers)
+    log_options = Keyword.get(opts, :log_options)
+    retry_options = Keyword.get(opts, :retry_options)
```

Comment:
Considering that the arguments could not be nil in the typespec of the previous version, I assume that all these options should have a default value.

Moreover, it could make sense to put here the default values that are in `BridgeEx.Graphql`, so who directly uses this module has the same defaults.

Reply:
I moved the default values to the client, but I still had to leave most of the code in `BridgeEx.Graphql` since it's needed to get values from the macro and so on.
Still, now if someone uses the client it will actually supply some default args its options

Let me know what you think

Reply:
It was exactly what I meant 👍